### PR TITLE
Fixed some issues from PVS-Studio report (step2)

### PR DESCRIPTION
**Pull request by igorkorsukov**
Resolves: from PVS-Studio report 

We analyzed the MuseScore code using PVS-Studio, get a report with warnings and errors.
Fixed some issues, step 2

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue

### CODE REVIEW

File: mscore/workspacedialog.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ void MuseScore::createNewWorkspace()
 
 void MuseScore::editWorkspace()
       {
-      if (!WorkspacesManager::currentWorkspace() && !WorkspacesManager::currentWorkspace()->readOnly())
+      if (WorkspacesManager::currentWorkspace() && !WorkspacesManager::currentWorkspace()->readOnly())
```
Comment (dmitrio95) [inline mscore/workspacedialog.cpp]: With this correction this check will disable editing workspace if it is not read-only while it seems like the opposite should happen. Could this check perhaps be intended to be like this?
```
if (!WorkspacesManager::currentWorkspace() || WorkspacesManager::currentWorkspace()->readOnly())
      return;
```

File: mscore/capella.cpp [hunk 2]
```diff
@@ -616,7 +616,8 @@ static Fraction readCapVoice(Score* score, CapVoice* cvoice, int staffIdx, const
                         if (tuplet) {
                               if (++nTuplet >= tupletCount) {
                                     tick = tupletTick + tuplet->actualTicks();
-                                    tuplet = 0;
+                                    delete tuplet;
```
Comment (AntonioBL) [inline mscore/capella.cpp]: I don't think this is correct. "tuplet" is a pointer to the newly created tuplet and it is reset to 0 when the tuplet is full so that if a new tuplet is found in the Capella file, a new tuplet in MuseScore score is created (see line 562). But the tuplet to which the pointer was pointing at is part of the score, so I don't think it should be deleted.

File: mscore/capella.cpp [hunk 3]
```diff
@@ -796,7 +797,8 @@ static Fraction readCapVoice(Score* score, CapVoice* cvoice, int staffIdx, const
...
```
Comment (AntonioBL) [inline mscore/capella.cpp]: Same as above

File: mscore/capella.cpp (hunk 2 above)
Comment (dmitrio95) [reply mscore/capella.cpp]: By the way, Capella tests fail on this PR.

**Comment by dmitrio95** [comment]
Also the test created for [this crash](https://musescore.org/en/node/295107) has failed, not sure about the reason though: https://travis-ci.org/github/musescore/MuseScore/jobs/664879918#L7454

File: mscore/capella.cpp (hunk 2 above)
Comment (igorkorsukov) [reply mscore/capella.cpp]: Unobvious code, the static analyzer decided that there was a memory leak.
Thanks, I’ll think about how to make the analyzer not think that there is a problem.

File: mscore/workspacedialog.cpp (hunk 1 above)
Comment (igorkorsukov) [reply mscore/workspacedialog.cpp]: Very confusing code, from the point of view of a static analyzer, there may be an appeal to an invalid pointer. Thank you, I will look what should be.

**Comment by AntonioBL** [comment]
I think the problem is this change https://github.com/musescore/MuseScore/pull/5850/files#diff-a7fddef1509bdc1ec91a110d8e6f5136R168
as seen in the log here: https://travis-ci.org/github/musescore/MuseScore/jobs/664879918#L7472
The fact that editData.element is not null is checked some lines before (line 161); endEdit is probably needed by the actions involved. The next check is because some endEdit (for some editData) change editData.element to 0, while others probably don't.

**Comment by igorkorsukov** [comment]
> I think the problem is this change https://github.com/musescore/MuseScore/pull/5850/files#diff-a7fddef1509bdc1ec91a110d8e6f5136R168
> as seen in the log here: https://travis-ci.org/github/musescore/MuseScore/jobs/664879918#L7472
> The fact that editData.element is not null is checked some lines before (line 161); endEdit is probably needed by the actions involved. The next check is because some endEdit (for some editData) change editData.element to 0, while others probably don't.

Oh, unexpectedly. Thanks, I’ll look in more detail and fix it.

File: mscore/workspacedialog.cpp (hunk 1 above)
Comment (Spire42) [reply mscore/workspacedialog.cpp]: > With this correction this check will disable editing workspace if it is not read-only while it seems like the opposite should happen. Could this check perhaps be intended to be like this?
> 
> ```
> if (!WorkspacesManager::currentWorkspace() || WorkspacesManager::currentWorkspace()->readOnly())
>       return;
> ```

Yes, that definitely seems to be the intention.

File: mscore/capella.cpp (hunk 2 above)
Comment (igorkorsukov) [reply mscore/capella.cpp]: @AntonioBL I see if the item will be added, depending on the condition
```
if (!o->invisible || voice == 0) {
```
Accordingly, a tuplet may not be added and there will be a memory leak, right?
Maybe you know this situation is typical or exceptional?

Comment (igorkorsukov) [reply mscore/capella.cpp]: @AntonioBL @dmitrio95 I ask you to look, is it correct now?

File: mscore/workspacedialog.cpp (hunk 1 above)
Comment (igorkorsukov) [reply mscore/workspacedialog.cpp]: Changed the code

File: mscore/workspacedialog.cpp [hunk 4]
```diff
@@ -37,8 +37,12 @@ void MuseScore::createNewWorkspace()
 
 void MuseScore::editWorkspace()
       {
-      if (!WorkspacesManager::currentWorkspace() && !WorkspacesManager::currentWorkspace()->readOnly())
+      if (!WorkspacesManager::currentWorkspace())
```
Comment (Spire42) [inline mscore/workspacedialog.cpp]: This is logically correct now, but what's the point of splitting it out into two separate `if` statements?

Comment (igorkorsukov) [reply mscore/workspacedialog.cpp]: To make it clear.

Comment (igorkorsukov) [reply mscore/workspacedialog.cpp]: And I think the condition `if (WorkspacesManager::currentWorkspace()) should be assert, but not sure yet

File: mscore/capella.cpp (hunk 2 above)
Comment (AntonioBL) [reply mscore/capella.cpp]: I think so. For practically all of the cases, that tuplet will not be deleted (i.e. it will be filled by chord or rest elements).
In principle, by looking at the code, tuplet->elements() could be empty only for tuplets made of invisible rests. This could be a problem only if the rests are voice 0 rests, but it is a remote case, and the original import code is wrong as well.
I may be lazy, but I don't think it is worth the effort to consider also that remote corner case in the code, considering that tuplet import in capella files is limited to simple tuplets only (complex tuplets, especially for old capella files, are encoded in a way which makes it difficult to analyze tuplet duration and its elements a priori, if I remember correctly).
In the years, there were no requests for complex tuplet import.
In summary: for me it is ok. Thank you. 👍
